wcfsetup/install/files/lib/data/file/File.class.php

```diff
@@ -240,4 +242,30 @@ public static function getSafeFileExtension(string $mimeType, string $filename):
 
         return 'bin';
     }
+
+    #[\Override]
+    public function getImageData(?int $minWidth = null, ?int $minHeight = null): ?ImageData
+    {
+        if (!$this->isImage()) {
+            return null;
+        }
+
+        if ($minWidth !== null || $minHeight !== null) {
+            $thumbnails = $this->getThumbnails();
+            usort($thumbnails, fn(FileThumbnail $a, FileThumbnail $b) => $a->width <=> $b->width);
+
+            foreach ($thumbnails as $thumbnail) {
+                if ($minWidth !== null && $minWidth > $thumbnail->width) {
+                    continue;
+                }
+                if ($minHeight !== null && $minHeight > $thumbnail->height) {
+                    continue;
+                }
+
+                return new ImageData($thumbnail->getLink(), $thumbnail->width, $thumbnail->height);
+            }
+        }
+
+        return new ImageData($this->getLink(), $this->width, $this->height);
```

Comment:
This will unconditionally return the image even if it does not satisfy the `$minWidth` and/or `$minHeight` constraints.